android/src/main/java/com/tanguyantoine/react/MusicControlModule.java

```diff
@@ -212,33 +203,47 @@ synchronized public void setNowPlaying(ReadableMap metadata) {
         nb.setContentInfo(album);
         nb.setColor(notificationColor);
 
-        if(artwork != null) {
-            final String artworkUrl = artwork;
-            final boolean artworkLocal = localArtwork;
-
-            artworkThread = new Thread(new Runnable() {
-                @Override
-                public void run() {
-                    Bitmap bitmap = loadArtwork(artworkUrl, artworkLocal);
-                    md.putBitmap(MediaMetadataCompat.METADATA_KEY_ART, bitmap);
-                    nb.setLargeIcon(bitmap);
-
-                    session.setMetadata(md.build());
-                    notification.show(nb, isPlaying);
-                    artworkThread = null;
-                }
-            });
-            artworkThread.start();
+        if (metadata.hasKey("artwork")) {
+          setArtwork(metadata);
         } else {
-            md.putBitmap(MediaMetadataCompat.METADATA_KEY_ART, null);
-            nb.setLargeIcon(null);
+          md.putBitmap(MediaMetadataCompat.METADATA_KEY_ART, null);
+          nb.setLargeIcon(null);
         }
 
         session.setMetadata(md.build());
         session.setActive(true);
         notification.show(nb, isPlaying);
     }
 
+    private void setArtwork(ReadableMap metadata) {
+      String artwork = null;
+      boolean localArtwork = false;
+
+      if(metadata.getType("artwork") == ReadableType.Map) {
+          artwork = metadata.getMap("artwork").getString("uri");
+          localArtwork = true;
+      } else {
+          artwork = metadata.getString("artwork");
+      }
+
+      final String artworkUrl = artwork;
+      final boolean artworkLocal = localArtwork;
+
+      artworkThread = new Thread(new Runnable() {
+          @Override
+          public void run() {
+              Bitmap bitmap = loadArtwork(artworkUrl, artworkLocal);
+              md.putBitmap(MediaMetadataCompat.METADATA_KEY_ART, bitmap);
+              nb.setLargeIcon(bitmap);
+
+              session.setMetadata(md.build());
+              notification.show(nb, isPlaying);
+              artworkThread = null;
+          }
+      });
+      artworkThread.start();
+    }
```

Comment:
It's called in setNowPlaying and updatePlayback, so that you can update the artwork if you want to change it while the audio is currently playing.  I use this to load a generic artwork first and then load one over the network later.